src/openrct2/world/Sprite.cpp

```diff
@@ -54,8 +54,8 @@ constexpr size_t GetSpatialIndexOffset(int32_t x, int32_t y)
         x = std::clamp(x, 0, 0xFFFF);
         y = std::clamp(y, 0, 0xFFFF);
 
-        int16_t flooredX = floor2(x, 32);
-        uint8_t tileY = y >> 5;
+        const auto flooredX = floor2(x, 32);
+        const auto tileY = y >> 5;
         index = (flooredX << 3) | tileY;
```

Comment:
With the current code, for every tile X increases the index by 256, and Y increases it by 1. Just adding their tile coords and multiply X by the max size is so much simpler. It would be much clearer what this is trying to accomplish by simply adding (x * max map size + y) together, where x and y are tile coords.

Also, when the max map size is not a power of 2 anymore (NSF), this would skip certain parts in the array. With the multiplication + addition you don't have that problem.

Reply:
```suggestion
        const auto tileX = x / 32;
        const auto tileY = y / 32;
        index = x * MAXIMUM_MAP_SIZE_TECHNICAL + tileY;
```

Reply:
*whacks Hielke with the keyboard*
We have spent a lot of effort to get rid of naked division by 32 and such things. Please don’t reintroduce them. Use TileCoords(CoordsXY(x, y)).

Reply:
The divisions were already there, just not as visible. `floor2(x, 32)` then shifting left by 3 and `y >> 5` are essentially divisions by 32 in disguise!